File: rpg.js
class Character {
    constructor(_name, _gender, _race, _charClass, _weapon) {
        this.name = _name
        this.gender = _gender
        this.race = _race
        this.charClass = _charClass
        this.weapon = _weapon

        this.skillSlot = 2
        this.exp = 0
        this.lvl = 1

        this.signatureSkill;
        if(this.charClass === 'rogue') {
            this.signatureSkill = 'backstab'
        } else if(this.charClass === 'wizard') {
            this.signatureSkill = 'fireball'
        }
    }

    greeting() {
        let something = 0
        // console.log('Hello, I am ' + this.name + '. I am a ' + this.race + ' ' + this.charClass)
        console.log(`Hello, I am ${this.name}. I am a ${this.race} ${this.charClass}`)
    }

    home() {
        let homeLocation = new Location('sunny', 'mountain', 'USA', 'tampa')
        if(this.lvl === 1) {
            console.log(this.name + ' starts at ' + homeLocation.country + ' ' + homeLocation.city)
        }
        else {
            console.log(this.name + ' has moved on...')
        }
    }

    stats() {
        console.log('=============================')
        console.log('STATS FOR: ' + this.name)
        console.log('CLASS: ' + this.charClass)
        console.log('RACE: ' + this.race)
        console.log('=============================')
    }
    
}

class Location {
    constructor(_climate, _terrain, _country, _city) {
        this.climate = _climate
        this.terrain = _terrain
        this.country = _country
        this.city = _city

    }

}

let theRogue = new Character('Zap Branigan', 'male', 'human', 'rogue', 'dagger');
let theWizard = new Character('Gandolf', 'male', 'human', 'wizard', 'wand');

function something() {

}

theRogue.greeting()
// theWizard.greeting()

// console.log(theRogue.signatureSkill)
// console.log(theWizard.signatureSkill)

theRogue.stats()
theWizard.stats()

File: rpg.go
package main

import "fmt"

///////////////////////////////////////////////////////////////////////////////
// CLASS OR STRUCT FOR CHARACTERS
///////////////////////////////////////////////////////////////////////////////
type Character struct {
	name string
	gender string
	race string
	charClass string
	weapon string
	
}

type Location struct {
	climate string
	terrain string
	country string
	city string
}

type DefaultAttributes struct {
	skillSlot int
	exp int
	level int
}

///////////////////////////////////////////////////////////////////////////////
// SETTING CONSTRUCTORS FOR CLASSES / INSTANTIATING OBJECTS
///////////////////////////////////////////////////////////////////////////////
func NewCharacter(_name, _gender, _race, _charClass, _weapon string) *Character {
	return &Character{ _name, _gender, _race, _charClass, _weapon }
}

func StartingAttributes(_skillSlot, _exp, _level int) DefaultAttributes {
	return DefaultAttributes{ 2, 0, 1 }
}

func Locale(_climate, _terrain, _country, _city string) *Location {
	return &Location{ _climate, _terrain, _country, _city }
}

///////////////////////////////////////////////////////////////////////////////
// METHODS / FUNCTION
///////////////////////////////////////////////////////////////////////////////
func (c Character) home() {
	home := *Locale("sunny", "mountain", "mordor", "moreDoor")
	fmt.Println(c.name + " starts at " + home.country + " in the great city of " + home.city)
}

func (c Character) applySkill() *string {
	signatureSkill := ""
	if c.charClass == "rogue" {
		signatureSkill = "Backstab"
	} else if c.charClass == "wizard" {
		signatureSkill = "FireBall"
	}
	return &signatureSkill
}

func (c Character) greeting() {
	fmt.Println("Hello, my name is " + c.name)
}

func (c Character) specialAttack() {
	fmt.Println(c.name + " uses " + *c.applySkill() + "!")
}

///////////////////////////////////////////////////////////////////////////////
// MAIN FUNCTION
///////////////////////////////////////////////////////////////////////////////
func main() {
	theRogue := Character{ "Zap Branigan", "male", "human", "rogue", "dagger" }
	theRogue.home()
	// theRogue.greeting()
	// theRogue.specialAttack()

	theWizard := Character{ "Henry Plopper", "male", "human", "wizard", "staff" }
	theWizard.home()
	// theWizard.greeting()
	// theWizard.specialAttack()

	
}